### ISSUE TITLE
Updating the HTML area to also include a Visual Editor.

### ISSUE THREAD

**Issue by paaljoachim** [CalderaWP/Caldera-Forms#1436]
Think TinyMCE
Visual mode - Text/HTML mode.

It would be good to have a visual mode area so one can easily paste in text and have the same formatting, or easily click the b or heading options in the visual editor in the backend.

I just pasted in text into the HTML field and got this:

![screen shot 2017-03-26 at 19 42 57](https://cloud.githubusercontent.com/assets/5323259/24333684/c0d60cfa-125c-11e7-9a1b-0683a2e7609a.png)

That means I have to clean it up to be sure that it looks like it should, even though it looks nice on the right side in the html box it is a mess on the left side.

So once again. It would be nice with a visual paragraph area similar to the html area but with the tinymce kind of buttons using the editor you decided on for the rich text frontend editor.

**Comment by paaljoachim** [comment]
Please add the frontend tinymce field also to the backend as another option to use instead of the HTML field.

**Comment by regiside** [comment]
I agree, Kind of blown away that such a feature rich form plugin doesn't give you any way to style text in the forms without knowing html. Just tried to teach one of my friends how to use the plugin and they were totally overwhelmed when they saw the HTML. 

I tried to add a javascript workaround implementation. But it's quite difficult since the textarea is dynamically created and is not present on initial page load. 

Are there any plans to offer a rich text editor for form creators?

**Comment by regiside** [comment]
Here's a relatively easy js workaround until this implemented in the Caldera:

Load this Javascript in your admin area:
`
jQuery( document ).ready(function() {
    
    tinymce.init({
        selector: '#layout-config-panel .caldera-config-field-setup textarea.block-input',  // change this value according to your HTML
        branding: false,
        plugins: [ 'image' ],
        block_formats: 'Paragraph=p; Header 1=h1; Header 2=h2; Header 3=h3',
        menubar: false,
        toolbar: 'styleselect | bold italic | alignleft aligncenter alignright | outdent indent link image',
        elementpath: false,
        setup: function (editor) {
            editor.on('change', function (editor) {
                var currentEditorID = '#' + tinyMCE.activeEditor.editorContainer.id;
                //jQuery(currentEditorID).next('textarea').html( tinymce.activeEditor.getContent() );
                tinyMCE.triggerSave();
            });
        }
    });
});
`

Or you can just use this plugin I created:
[caldera-forms-html-wysiwyg-editor.zip](https://github.com/CalderaWP/Caldera-Forms/files/4291086/caldera-forms-html-wysiwyg-editor.zip)

Notes:
The text doesn't update on the left pane unless you save the form, or to quickly see the changes, you can check and uncheck the "Email Summary?" box beneath the editor.

For now there's no magic tags helper. You could still use them you just have to remember them manually. (As far as I know Magic tags in HTML don't update live on the site anyway, so they're really only useful for Email Summary.)
